tools/tune/eq/gui.m

```diff
@@ -0,0 +1,476 @@
+      "userdata", struct(
+          "index", index, ...
+          "channel", channel, ...
+          "prefix", "Q", ...
+          "EQ", EQ, ...
+          "main_panes", main_panes), ...
+      "visible", "off", ...
+      "callback", @slider_callback);
+
+  EQ(index, channel).freq.label = uicontrol( ...
+      "style", "text", ...
+      "parent", p, ...
+      "position", [0 90 40 30], ...
+      "visible", "off", ...
+      "string", "Freq");
+  EQ(index, channel).freq.edit = uicontrol( ...
+      "style", "edit", ...
+      "parent", p, ...
+      "position", [40 90 90 30], ...
+      "string", "350", ...
+      "userdata", struct( ...
+          "index", index, ...
+          "channel", channel, ...
+          "prefix", "freq", ...
+          "EQ", EQ, ...
+          "main_panes", main_panes), ...
+      "visible", "off", ...
+      "callback", @edit_callback);
+  EQ(index, channel).freq.units = uicontrol( ...
+      "style", "text", ...
+      "parent", p, ...
+      "position", [140 90 17 30], ...
+      "visible", "off", ...
+      "string", "Hz");
+  EQ(index, channel).freq.slider   = uicontrol( ...
+      "style", "slider", ...
+      "parent", p, ...
+      "position", [160 90 200 30], ...
+      "min", 1, ...
+      "max", nyquist_f, ...
+      "value", 350, ...
+      "userdata", struct(
+          "index", index, ...
+          "channel", channel, ...
+          "prefix", "freq", ...
+          "EQ", EQ, ...
+          "main_panes", main_panes), ...
+      "visible", "off", ...
+      "callback", @slider_callback);
+
+  EQ(index, channel).type.label = uicontrol( ...
+      "style", "text", ...
+      "parent", p, ...
+      "position", [0 130 40 30], ...
+      "visible", "off", ...
+      "string", "Type");
+  EQ(index, channel).type.dropdown = uicontrol( ...
+      "style", "popupmenu", ...
+      "parent", p, ...
+      "position", [40 130 240 30], ...
+      "string", eq_strings, ...
+      "value", eq.PEQ_PN2, ...
+      "userdata", struct( ...
+          "index", index, ...
+          "channel", channel, ...
+          "EQ", EQ, ...
+          "main_panes", main_panes), ...
+      "visible", "off", ...
+      "callback", @dropdown_callback);
+  EQ(index, channel).enable_switch = uicontrol( ...
+      "style", "checkbox", ...
+      "parent", p, ...
+      "position", [350 130 30 30], ...
+      "value", 0, ...
+      "userdata", struct(
+          "index", index, ...
+          "channel", channel, ...
+          "EQ", EQ, ...
+          "main_panes", main_panes), ...
+      "callback", @enable_callback);
+end
+
+function push_config(h, evnt)
+  udata = get(h, "userdata");
+  window = udata.menu_window;
+  EQ = udata.EQ;
+  ip = get(window.ip_box, "string");
+  device = str2num(get(window.device_box, "string"));
+  alsa_num = str2num(get(window.alsa_box, "string"));
+  target = struct("ip", ip, "device", device, "control", alsa_num);
+
+  temp_file = tempname();
+
+  eq_left = eq_defaults();
+  eq_left.enable_iir = 1;
```

Comment:
Due to to the scaling choise in eq_defaults() for constant loudness doing setting from GUI for a high-pass can create a very loud annoying sound.  A more safe choice if not GUI defined could be to set e.g.

```
  eq_left.norm_type = 'peak';
  eq_left.norm_offs_db = 0;

```
And the same for right channel. If you have preferences for normalization approach please let me know and I'll add it to the core functions!

Reply:
@dgreid do you have any preferences as to normalization types?

Reply:
This change is added but leaving unresolved for @dgreid to comment on.